src/mc_control/generic_gripper.cpp

```diff
@@ -279,4 +299,46 @@ double Gripper::opening() const
   return std::accumulate(percentOpen.begin(), percentOpen.end(), 0.0) / static_cast<double>(percentOpen.size());
 }
 
+bool Gripper::complete() const
+{
+  bool res = (targetQ == nullptr);
+  if(!res)
+  {
+    res = true;
+    for(size_t i = 0; i < actualQ.size(); ++i)
+    {
+      res = res && overCommandLimit[i];
+    }
+  }
+  return res;
+}
+
+void Gripper::addToGUI(mc_rtc::gui::StateBuilder & gui, std::vector<std::string> category)
+{
+  using namespace mc_rtc::gui;
+  std::vector<std::string> cat = category;
+  cat.push_back(name);
+  gui.addElement(
+      cat, Button("Open", [this]() { setTargetOpening(1); }), Button("Close", [this]() { setTargetOpening(0); }),
+      NumberInput("Opening percentage", [this]() { return opening(); }, [this](double op) { setTargetOpening(op); }),
+      NumberInput("Percentage VMAX", [this]() { return percentVMAX(); }, [this](double op) { percentVMAX(op); }));
```

Comment:
We can use `NumberSlider` for "Percentage VMAX" and "Opening percentage" here

Reply:
Good call, looks much nicer ;)